api/rest/restapi.go

```diff
@@ -925,16 +972,7 @@ func (api *API) recoverHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func (api *API) parseCidOrError(w http.ResponseWriter, r *http.Request) types.PinSerial {
-	vars := mux.Vars(r)
-	hash := vars["hash"]
-
-	_, err := cid.Decode(hash)
-	if err != nil {
-		api.sendResponse(w, http.StatusBadRequest, errors.New("error decoding Cid: "+err.Error()), nil)
-		return types.PinSerial{Cid: ""}
-	}
-
+func (api *API) parsePinSerial(w http.ResponseWriter, r *http.Request, hash string) types.PinSerial {
```

Comment:
I think this can become `parsePinOptions` and return `*types.PinOptions`.

Disregard MaxDepth. We can handle that somewhere else. The user should only be able to manage PinOptions.

Reply:
It's not just MaxDepth. PinType, Allocations and Reference are also required. On removing those, we would fail checkPinType
https://github.com/ipfs/ipfs-cluster/blob/a244af9a13558d59ac3670e266516fa2eb95b26f/cluster.go#L925-L961

Reply:
No, MaxDepth, PinType and Allocations are not user-dependant. You do not need to parse them from the query, they just need to take defaults. We'll deal with allocations in the other PR.